library/src/main/java/com/bumptech/glide/request/transition/ViewPropertyTransition.java

```diff
@@ -12,6 +13,7 @@
 public class ViewPropertyTransition<R> implements Transition<R> {
 
   private final Animator animator;
+  private WeakReference<ViewAdapter> viewAdapter;
```

Comment:
I don't think there's a need for a weak reference here. If there is, can you add a comment explaining why it's necessary?

Reply:
As said in the issue discussion I'm not sure to fully understand all internals of Glide (Actually I'm sure I don't :p) so I chose the safe way to avoid hard references as I don't know if there's case where the transition can last too long.

Reply:
:) A hard reference here should be fine then, especially since you're adding an API to cancel and end the transition in clear anyway.